Automattic/jetpack
# Update babel-eslint → @babel/eslint-parser

#### Changes proposed in this Pull Request:

[babel-eslint](https://www.npmjs.com/package/babel-eslint) has been renamed to [@babel/eslint-parser](https://www.npmjs.com/package/@babel/eslint-parser). Let's make that change.

That also requires some config changes to point @babel/eslint-parser at the babel config in each project, as the new package uses the actual babel config to know what lints to check for.

This also makes a few other fixes related to eslint:

* Add some missing built files to `.eslintignore`.
* Include `.cjs` files in linting.
* We're getting an error about "React version was set to "detect" in eslint-plugin-react settings, but the "react" package is not installed." Fix that by only loading react rules when processing projects that use react.
* Move the eslint deps to a workspace, so as to not clutter the monorepo root with them.

#### Jetpack product discussion


None.

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Is CI happy?
* Can you run eslint and eslint-changed like you could before? Note that you may need to do `jetpack clean all node_modules && pnpm install` after checking this out to fix the `eslint` symlink.
